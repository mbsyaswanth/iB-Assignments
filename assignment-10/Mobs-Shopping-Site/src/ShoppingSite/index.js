import React, { Component } from "react";

class ShoppingSite extends Component {
  render() {
    return <div />;
  }
}

export default ShoppingSite;
