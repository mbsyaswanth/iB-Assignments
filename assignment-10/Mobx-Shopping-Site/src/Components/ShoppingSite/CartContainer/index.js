import React, { Component } from "react";
import { observer } from "mobx-react";
import { action, computed, observable } from "mobx";
import CartItem from "../CartContainer/CartItem";

@observer
class CartContainer extends Component {
  @observable isClicked = false;
  @action cartClick = () => {
    this.isClicked = !this.isClicked;
  };

  @computed get loadIcon() {
    let cart = (
      <img alt="carticon" className="cart-image" src="assets/cart.png" />
    );
    return this.isClicked ? "x" : cart;
  }

  render() {
    return (
      <div className={"cart-container " + (this.isClicked ? "cart-show" : "")}>
        <div onClick={this.cartClick} className="cart-toggle">
          {this.loadIcon}
          <span className={this.isClicked ? "display-none" : "cart-count"}>
            {this.props.store.cartItemsCount}
          </span>
        </div>
        <div className="cart-items-container">
          <div className="cart-items-inner-wrap">
            <div className="cart-icon-in-cart">
              <div className="cart-toggle">
                <img
                  alt="carticon"
                  className="cart-image in-cart-image"
                  src="assets/cart.png"
                />
                <span className="cart-count">
                  {this.props.store.cartItemsCount}
                </span>
              </div>
              <div className="cart-text">Cart</div>
            </div>
            <div className="cart-items-wrapper">
              {this.props.store.cart.map(item => {
                return (
                  <CartItem
                    key={item.id}
                    id={item.itemId}
                    product={this.props.store.getProduct(item.itemId)}
                    store={this.props.store}
                    quantity={item.quantity}
                  />
                );
              })}
            </div>
          </div>
          <div className="cart-subtotal">
            <div className="sub-total">
              <span className="sub-total-text">SUBTOTAL</span>
              <span className="sub-total-count">
                <span>$</span>
                {this.props.store.totalCartValue}
              </span>
            </div>
            <div className="checkout">CHECKOUT</div>
          </div>
        </div>
      </div>
    );
  }
}

export default CartContainer;
