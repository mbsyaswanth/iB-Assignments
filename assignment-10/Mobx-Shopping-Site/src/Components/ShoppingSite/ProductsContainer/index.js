import React, { Component } from "react";
import Product from "./Product";
import { observer } from "mobx-react";
import { action, computed, observable } from "mobx";

@observer
class ProductsContainer extends Component {
  render() {
    return (
      <div className="products-container">
        {this.props.store.filteredProducts.map(product => {
          return (
            <Product
              key={product.id}
              product={product}
              store={this.props.store}
            />
          );
        })}
      </div>
    );
  }
}

export default ProductsContainer;
