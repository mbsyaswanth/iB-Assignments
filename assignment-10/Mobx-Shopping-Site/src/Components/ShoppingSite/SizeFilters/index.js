import React, { Component } from "react";

import { observer } from "mobx-react";
import { action, computed, observable } from "mobx";

@observer
class SizeFilters extends Component {
  handleClick = event => {
    this.props.onFilterClick(event.target.value);
  };

  render() {
    return (
      <div className="size-filters-container">
        <div>
          <div className="size-filter-row1">
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("XS")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="XS"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("S")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="S"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("M")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="M"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("ML")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="ML"
            />
          </div>
          <div>
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("L")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="L"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("XL")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="XL"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("XXL")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="XXL"
            />
          </div>
        </div>
      </div>
    );
  }
}

export default SizeFilters;
