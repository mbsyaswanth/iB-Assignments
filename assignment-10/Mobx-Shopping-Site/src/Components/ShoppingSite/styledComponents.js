import styled from "styled-components";

export const ShoppingSiteContainer = styled.div`
  padding-top: 50px;
  display: flex;
`;

export const FiltersSidebar = styled.div`
  width: 15%;
  padding-right: 5px;
`;

export const SizesHeading = styled.h3`
  padding-left: 40px;
`;

export const MainContentContainer = styled.div`
  width: 85%;
`;

export const SortBy = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 10px;
`;
