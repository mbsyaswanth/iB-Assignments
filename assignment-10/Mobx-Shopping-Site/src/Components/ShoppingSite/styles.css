.shopping-site-container {
  padding-top: 50px;
  display: flex;
}
/* TODO: seperate styles for each component */
.filters-sidebar {
  width: 15%;
  padding-right: 5px;
}

.main-content-container {
  width: 85%;
}

.sort-by {
  display: flex;
  justify-content: space-between;
  padding: 10px;
}

.product-container {
  position: relative;
  padding: 10px;
  margin: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 333px;
  background-color: white;
  transition: 0.2s;
}

.product-container:hover {
  box-shadow: 0px 0px 2px 1px grey;
}

.free-shipping {
  position: absolute;
  top: 10px;
  right: 10px;
  color: white;
  padding: 2px;
  background-color: black;
}

.product-info {
  display: flex;
  flex-direction: column;
  height: 160px;
  justify-content: space-evenly;
  align-items: center;
  margin-top: 10px;
}

.product-title {
  font-size: 1.2rem;
  height: 2.4rem;
}

.line {
  width: 30px;
  height: 2px;
  background-color: goldenrod;
}

.product-price {
  font-weight: 600;
  font-size: 1.6rem;
}

.product-installments {
  color: grey;
  font-weight: 600;
}

.product-cart-btn {
  width: 333px;
  padding: 10px;
  font-size: 1.1rem;
  color: white;
  background-color: black;
  font-weight: 500;
  border-style: none;
  cursor: pointer;
  transition: 0.2s;
}

.product-cart-btn:focus {
  outline: none;
}

.product-cart-btn:active {
  border-style: none;
}

.product-container:hover .product-cart-btn {
  background-color: goldenrod;
}

.display-none {
  display: none;
}

.products-container {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.size-filter {
  padding: 5px;
  margin: 5px;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  background-color: rgba(219, 219, 219, 0.384);
  text-align: center;
  border-style: none;
  transition: 0.3s;
  cursor: pointer;
}
.size-filter:focus {
  outline: none;
}

.size-filters-container {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.size-filter-row1 {
  padding-bottom: 5px;
}

.size-filter-select {
  color: white;
  background-color: black;
}

.size-filter:hover {
  box-shadow: 0px 0px 1px 2px black;
}

.size-filter:active {
  border-style: none;
}

.sizes-heading {
  padding-left: 40px;
}

.cart-image {
  height: 60px;
  width: 60px;
}

.cart-container {
  position: fixed;
  z-index: 999;
  display: flex;
  right: -40vw;
  transition: 0.5s;
}
.cart-toggle {
  position: relative;
  height: 60px;
  width: 60px;
  background-color: rgb(8, 7, 7);
  cursor: pointer;
  color: white;
  font-size: 42px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.cart-items-container {
  height: 100vh;
  width: 40vw;
  background-color: rgb(8, 7, 7);
}

.cart-show {
  right: 0;
}

.cart-count {
  position: absolute;
  bottom: 5px;
  right: 5px;
  background-color: goldenrod;
  font-size: 12px;
  padding: 2px;
  border-radius: 50%;
}

.cart-subtotal {
  height: 23%;
  padding: 25px;
  font-size: 16px;
  box-shadow: 0px -5px 20px 1px black;
}
.sub-total {
  display: flex;
  justify-content: space-between;
}
.cart-items-inner-wrap {
  height: 77%;
  padding: 25px;
}
.sub-total-text {
  color: grey;
}

.sub-total-count {
  color: goldenrod;
}

.checkout {
  margin-top: 60px;
  color: white;
  background-color: black;
  padding: 15px;
  text-align: center;
  cursor: pointer;
}

.cart-icon-in-cart {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 15px;
}

.cart-text {
  font-size: 20px;
  color: white;
}

.in-cart-image {
  height: 100px;
  width: 100px;
}

/* cart item */

.cart-item {
  display: flex;
  position: relative;
  padding-top: 10px;
  padding-bottom: 5px;
  box-shadow: 0px -3px 4px 0px black;
  margin-bottom: 10px;
}

.cart-item-image {
  height: 100px;
}

.cart-item-info {
  padding: 10px;
  display: flex;
  justify-content: space-between;
}

.cart-item-info-wrapper {
  flex-basis: 100%;
  align-self: center;
}

.item-name {
  color: white;
  font-size: 1.3rem;
}

.item-price {
  display: flex;
  color: goldenrod;
  align-items: center;
}

.size-style,
.item-qty {
  color: grey;
}

.cross-mark {
  position: absolute;
  right: 5px;
  top: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 800;
  padding-right: 5px;
  cursor: pointer;
}

.cross-mark:hover .cart-item-info {
  text-decoration: line-through;
}
