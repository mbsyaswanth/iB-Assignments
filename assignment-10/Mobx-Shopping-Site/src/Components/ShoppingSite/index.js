import React, { Component } from "react";

import SizeFilters from "./SizeFilters";
import ProductsContainer from "./ProductsContainer";
import CartContainer from "./CartContainer";

import { orderBy } from "../../constants";
import {
  ShoppingSiteContainer,
  FiltersSidebar,
  SizesHeading,
  MainContentContainer,
  SortBy
} from "./styledComponents";
import { observer } from "mobx-react";

@observer
class ShoppingSite extends Component {
  handleOrderChange = event => {
    console.log("orderby triggered");
    this.props.store.updateOrderBy(event.target.value);
  };

  render() {
    return (
      <>
        <CartContainer store={this.props.store} />
        <ShoppingSiteContainer>
          <FiltersSidebar>
            <SizesHeading>Sizes:</SizesHeading>
            <SizeFilters
              onFilterClick={this.props.store.updateSizeFilters}
              sizes={this.props.store.sizes}
            />
          </FiltersSidebar>
          <MainContentContainer>
            <SortBy>
              <div>
                {this.props.store.filteredProductsCount} product(s) found
              </div>
              <div>
                Order by{"  "}
                <select
                  value={this.props.store.orderBy}
                  onChange={this.handleOrderChange}
                >
                  {/* TODO: make lh hl as constants */}
                  <option value="select">select</option>
                  <option value={orderBy.lh}>low to high</option>
                  <option value={orderBy.hl}>high to low</option>
                </select>
              </div>
            </SortBy>
            <ProductsContainer store={this.props.store} />
          </MainContentContainer>
        </ShoppingSiteContainer>
      </>
    );
  }
}

export default ShoppingSite;
