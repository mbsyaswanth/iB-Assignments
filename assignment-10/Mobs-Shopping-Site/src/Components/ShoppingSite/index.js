import React, { Component } from "react";

import SizeFilters from "./SizeFilters";
import "./styles.css";

class ShoppingSite extends Component {
  render() {
    return (
      <div className="shopping-site-container">
        <div className="filters-sidebar">
          <h3>Sizes:</h3>
          <SizeFilters />
        </div>
        <div className="main-content-container">
          <div className="sort-by">
            <div>16 products found</div>
            <div>
              Order by{"  "}
              <select>
                <option value="select">select</option>
                <option value="lh">low to high</option>
                <option value="hl">high to low</option>
              </select>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default ShoppingSite;
