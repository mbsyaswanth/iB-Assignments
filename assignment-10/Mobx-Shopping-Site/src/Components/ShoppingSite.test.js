import React from "react";
import { shallow } from "enzyme";
import ShoppingSite from "./ShoppingSite";
import Products from "../Store";

let store = new Products();

describe("ShoppingSite", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ShoppingSite store={store} debug />);

    expect(component).toMatchSnapshot();
  });
});
