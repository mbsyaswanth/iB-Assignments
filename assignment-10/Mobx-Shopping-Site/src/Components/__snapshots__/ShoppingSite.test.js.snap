// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingSite should render correctly in "debug" mode 1`] = `
<Fragment>
  <CartContainer
    store={
      Products {
        "cart": Array [],
        "getIndexInProducts": [Function],
        "getProduct": [Function],
        "loginStatus": Object {},
        "msg": "",
        "orderBy": "",
        "productFetchStatus": "LOADING",
        "products": Array [],
        "signUpStatus": Object {
          "status": "",
        },
        "sizes": Array [],
      }
    }
  />
  <styled.div>
    <styled.div>
      <styled.h3>
        Sizes:
      </styled.h3>
      <SizeFilters
        onFilterClick={[Function]}
        sizes={Array []}
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <div>
          0
           product(s) found
        </div>
        <div>
          Order by
            
          <select
            onChange={[Function]}
            value=""
          >
            <option
              value="select"
            >
              select
            </option>
            <option
              value="LOW TO HIGH"
            >
              low to high
            </option>
            <option
              value="HIGH TO LOW"
            >
              high to low
            </option>
          </select>
        </div>
      </styled.div>
      <ProductsContainer
        store={
          Products {
            "cart": Array [],
            "getIndexInProducts": [Function],
            "getProduct": [Function],
            "loginStatus": Object {},
            "msg": "",
            "orderBy": "",
            "productFetchStatus": "LOADING",
            "products": Array [],
            "signUpStatus": Object {
              "status": "",
            },
            "sizes": Array [],
          }
        }
      />
    </styled.div>
  </styled.div>
</Fragment>
`;
